cli/lib/upgrade-kvm.js

```diff
@@ -0,0 +1,90 @@
+"use strict";
+
+const pem = require("pem");
+const util = require("util");
+const debug = require("debug")("jwkrotatekey");
+const request = require("request");
+
+
+function generateCredentialsObject(options) {
+    if (options.token) {
+        return {
+            "bearer": options.token
+        };
+    } else {
+        return {
+            user: options.username,
+            pass: options.password
+        };
+    }
+}
+
+const UpgradeKVM = function () {
+
+}
+
+module.exports = function () {
+  return new UpgradeKVM();
+}
+
+UpgradeKVM.prototype.upgradekvm = function upgradekvm(options, cb) {
+
+    options.baseuri = options.baseuri || "https://api.enterprise.apigee.com";
+    options.kvm = 'microgateway';
+    options.kid = '1';
+    options.virtualhost = options.virtualhost || 'secure';    
+
+    var publicKeyURI = util.format('https://%s-%s.apigee.net/edgemicro-auth/publicKey', options.org, options.env);
```

Comment:
This code also assumes Apigee SaaS instance.  Can we change this to handle on-prem customers?